src/pad_action_entry.rs

```diff
@@ -4,42 +4,48 @@
 
 use glib::translate::*;
 use gtk_sys;
+use std::ffi::CStr;
 use PadActionType;
 
-#[repr(C)]
-pub struct PadActionEntry(gtk_sys::GtkPadActionEntry);
+pub struct PadActionEntry {
+    type_: PadActionType,
+    index: i32,
+    mode: i32,
+    label: String,
+    action_name: String,
+}
 
 impl PadActionEntry {
     pub fn new(type_: PadActionType, index: i32, mode: i32, label: &str,
                action_name: &str) -> PadActionEntry {
         assert_initialized_main_thread!();
-        PadActionEntry(gtk_sys::GtkPadActionEntry {
-            type_: type_.to_glib(),
+        PadActionEntry {
+            type_,
             index,
             mode,
-            label: label.to_glib_none().0,
-            action_name: action_name.to_glib_none().0,
-        })
+            label: label.to_owned(),
+            action_name: action_name.to_owned(),
+        }
     }
 
     pub fn get_type(&self) -> PadActionType {
-        from_glib(self.0.type_)
+        self.type_
     }
 
     pub fn get_index(&self) -> i32 {
-        self.0.index
+        self.index
     }
 
     pub fn get_mode(&self) -> i32 {
-        self.0.mode
+        self.mode
```

Comment:
Shouldn't this be wrapped inside an enum or something to prevent users to change this value?

Reply:
How do you mean? The user sets this very value